frontend/src/services/effects.ts

```diff
@@ -217,6 +192,21 @@ export function withEffects(store: Store) {
       )
     )
 
+  /**
+   * When the user adds representative points,
+   * make sure that providers appear on top.
+   */
+  store
+    .on('representativePoints')
+    .subscribe(representativePoints => {
+      let temp_providers = store.get('providers')
+      if (representativePoints.length && temp_providers.length) {
+        store.set('providers')([])
```

Comment:
This feels a bit hacky. Can we consolidate the re-rendering of pointAs and pointBs into a single `store.on`? I think that would more elegantly solve the problem.

Reply:
What do you mean?

Reply:
And yes, it is a bit hacky

Reply:
I mean using `Observable.combineLatest`

Reply:
so that when either one changes we re-render both (in the correct order)

Reply:
We only need to re-render if providers are second though